blackbox/docs/sql/administration/privileges.txt

```diff
@@ -85,18 +135,67 @@ privileges to a user, namely ``DQL``, ``DML`` and ``DDL``.
 
 Privileges can be granted to multiple users in the same statement, like so::
 
-    cr> GRANT DDL TO wolfgang, will;
+    cr> GRANT DDL ON TABLE doc.books TO wolfgang, will;
     GRANT OK, 2 row affected (... sec)
 
+``DENY``
+........
+
+To deny a privilege to a user on the whole cluster use the
+:doc:`/sql/reference/deny` SQL statement, for example::
+
+    cr> DENY DDL from will;
+    DENY OK, 1 row affected (... sec)
+
+A privilege can be denied on a specific schema, like this::
+
+    cr> DENY DQL ON SCHEMA sys TO wolfgang;
+    DENY OK, 1 row affected (... sec)
+
+A privilege can be denied on a specific table, like this::
+
+    cr> DENY DQL ON TABLE doc.books TO wolfgang;
+    DENY OK, 1 row affected (... sec)
+
+If no schema is specified in the table ident, the current schema will be used.
+
+If a user with the username specified in the SQL statement does not exist
+the statement returns an error::
+
+    cr> DENY DDL FROM layla;
+    UserDoesNotExistException ...
+
+``DENY ALL`` or ``DENY ALL PRIVILEGES`` will deny all privileges to a user,
+it can be used like this::
+
+    cr> DENY ALL FROM wolfgang;
+    DENY OK, 3 row affected (... sec)
+
+Privileges can be denied from multiple users in the same statement,
+like so::
+
+    cr> DENY DDL FROM wolfgang, will;
+    DENY OK, 2 row affected (... sec)
+
 ``REVOKE``
-==========
+..........
 
-To revoke a privilege that was previously granted to a user use the
-:doc:`/sql/reference/revoke` SQL statement, for example::
+To revoke a privilege that was previously granted or denied to a user
+use the :doc:`/sql/reference/revoke` SQL statement, for example::
 
     cr> REVOKE DDL from will;
     REVOKE OK, 1 row affected (... sec)
 
+A privilege can be revoked on a specific schema, like this::
+
+    cr> REVOKE DQL ON SCHEMA sys FROM wolfgang;
+    REVOKE OK, 1 row affected (... sec)
```

Comment:
document the fact that revoke operates on a certain level in the hierarchy, even the one below which is on cluster. then you also have to document what the rowcount means